src/OrchardCore/OrchardCore.Roles.Abstractions/Extensions/SystemRoleProviderExtensions.cs

```diff
@@ -5,7 +5,11 @@ namespace OrchardCore.Roles;
 public static class SystemRoleProviderExtensions
 {
     public static async ValueTask<IRole> GetAdminRoleAsync(this ISystemRoleProvider systemRoleNameProvider)
-        => (await systemRoleNameProvider.GetSystemRolesAsync()).First();
+    {
+        var systemRoles = await systemRoleNameProvider.GetSystemRolesAsync();
+
+        return systemRoles.SingleOrDefault(role => role.RoleName.Equals(OrchardCoreConstants.Roles.Administrator, StringComparison.OrdinalIgnoreCase));
```

Comment:
this logic is invalid. you can't assume that the Administrator role is called Administrator. Please follow the suggestion I shared in my original feedback.

Reply:
Oops, seems I did something wrong here, that's why I preferred the position previously